java/shared/java/org/signal/libsignal/internal/Native.java

```diff
@@ -487,4 +488,6 @@ private Native() {}
   public static native void Username_Verify(byte[] proof, byte[] hash);
 
   public static native void UuidCiphertext_CheckValidContents(byte[] buffer);
+
+  public static native byte[] Grpc_SendMessage(String method, String urlFragment, byte[] body, Map<String, List<String>> headers);
```

Comment:
We can benefit from a higher-level abstraction that returns a real reply (a SignalRpcReply or similar).
e.g. see the classes in `java/shared/java/org/signal/libsignal/zkgroup/profiles` that contain a higher-level abstraction and those invoke the native methods on `Native.java`. They also process the result (e.g. a `byte[]`) and convert it into a meaningful object that can be returned to the caller.